codalab/worker/local_run/local_run_manager.py

```diff
@@ -364,6 +373,8 @@ def all_dependencies(self):
         """
         Returns a list of all dependencies available in this RunManager
         """
+        if self._shared_file_system:
+            return []
```

Comment:
Should probably add a comment here

Reply:
where `[]` means all bundles?  Might be confusing...

Reply:
It is kind of confusing, but short of changing the worker API or adding a special const value to report here like `ALL` (sounds smelly to me) there's no better solution here.

Also, since in the current model `shared_filesystem_worker` is both a server-side AND worker-side setting, the same server will never have both shared filesystem and regular workers, so the dependencies field will be essentially meaningless for shared filesystem setups. Given all this, I think it is safe to just report no dependencies here.

I should add a comment explaining this though.

Reply:
Could we return `None` at least so it can't be confused with no dependencies?

Reply:
Currently none of the upstream code is written to handle a worker that just doesn't report None for dependencies. Besides, even if we did that, the end result would be that the worker dependencies row for that worker would be empty on the database, so it wouldn't change much if we did that through reporting a None/null field on the checkin vs reporting no dependencies.

One thing that might be good to add is to update the worker filtering code to assume all dependencies are locally cached for all runs on shared fs workers, but I can make that change in the filtering logic rather than here.

Reply:
Just pushed the latter change I mentioned to the `filter_and_sort_workers` method.